### DISCUSSION TITLE
How to live at head

### DISCUSSION THREAD

**Discussion by stal12** [google/googletest#3691] [Community Help] [answered]
Hi, the readme now recommends to always use the latest commit in `main`.
The [Quickstart: CMake tutorial](https://google.github.io/googletest/quickstart-cmake.html) suggests to use the `FetchContent` CMake module, pointing at a certain git commit hash, and recommends to update it regularly.
Does this really mean that I'm supposed to manually change the url when a new commit is made on the repo? Is it bad to just put [https://github.com/google/googletest/archive/refs/heads/main.zip](https://github.com/google/googletest/archive/refs/heads/main.zip) instead?

**Comment by apcountryman** [comment, marked as answer]
There are a few potential issues with using a branch name instead of commit hashes:
- The fetch will fail if the branch is removed/renamed. This happened for some users recently when the master branch was renamed to main, but this isn't likely to be a problem going forward.
- Your builds will immediately break if breaking changes are introduced. If a specific git commit hash is used, the breaking changes can be addressed when you feel like updating.
- Builds cease to be reproducible once new commits are made to the fetched branch.

Living at head doesn't require that you update the URL every time a new commit is made, but updating it regularly is a good idea so that you get access to the latest bug fixes and enhancements, and avoid getting significantly out of sync which could require significant changes to your code base when you do get around to updating.
